packages/store-ui/src/organisms/Carousel/Carousel.tsx

```diff
@@ -0,0 +1,91 @@
+import type { PropsWithChildren } from 'react'
+import type { SwipeableProps } from 'react-swipeable'
+import React from 'react'
+
+import Button from '../../atoms/Button'
+import Icon from '../../atoms/Icon'
+import { RightArrowIcon, LeftArrowIcon } from './Arrows'
+import useCarousel from './hooks/useCarousel'
+import { useSlideVisibility } from './hooks/useSlideVisibility'
+import Bullets from '../../molecules/Bullets'
+
+export interface CarouselProps {
+  testId?: string
+  itemsPerPage?: number
+  swipeableConfigOverrides?: SwipeableProps
+}
+
+function Carousel({
+  testId = 'store-carousel',
+  itemsPerPage = 1,
+  swipeableConfigOverrides,
+  children,
+}: PropsWithChildren<CarouselProps>) {
+  const numberOfSlides = React.Children.count(children)
+
+  const { handlers, slide, carouselState, carouselDispatch } = useCarousel({
+    totalItems: numberOfSlides,
+    itemsPerPage,
+    swipeableConfigOverrides,
+  })
+
+  const { shouldRenderItem, isItemVisible } = useSlideVisibility({
+    itemsPerPage: carouselState.itemsPerPage,
+    currentSlide: carouselState.currentSlide,
+  })
+
+  return (
+    <section
+      data-store-carousel
+      data-testid={testId}
+      aria-label="carousel"
+      {...handlers}
+    >
+      <div data-carousel-track-container>
+        <div data-carousel-track>
+          {React.Children.map(children, (child, idx) => (
+            <div
```

Comment:
I think `shouldRenderItem` should also prevent the div from being rendered, shouldn't it?

Reply:
I asked myself this same question 😂 . But I opted for always rendering the wrapping div with `data-carousel-item` and `data-visible` (if that's the case) because then you can use CSS to set `width`s and `height`s for the slides, even before rendering them. This should be useful when trying to avoid sudden layout shifts since the actual slides will render into containers that already had their sizes allocated.

Reply:
Makes sense!